annotation/tests/test_tasks_services.py

```diff
@@ -0,0 +1,343 @@
+        "validation_user",
+        "expected_error_message_pattern",
+    ),
+    (
+        (
+            {1},
+            set(),
+            False,
+            True,
+            r"Missing `annotation_user_for_failed_pages` param",
+        ),
+        (
+            set(),
+            {2},
+            True,
+            False,
+            r"Missing `validation_user_for_reannotated_pages` param",
+        ),
+    ),
+)
+def test_validate_user_actions_missing_users(
+    failed: set,
+    annotated: set,
+    annotation_user: bool,
+    validation_user: bool,
+    expected_error_message_pattern: str,
+):
+    with pytest.raises(HTTPException) as excinfo:
+        validate_user_actions(
+            is_validation=True,
+            failed=failed,
+            annotated=annotated,
+            not_processed=set(),
+            annotation_user=annotation_user,
+            validation_user=validation_user,
+        )
+    assert re.match(expected_error_message_pattern, excinfo.value.detail)
+
+
+@pytest.mark.parametrize(
+    (
+        "failed",
+        "annotated",
+        "not_processed",
+        "expected_error_message_pattern",
+    ),
+    (
+        (
+            set(),
+            set(),
+            set(),
+            r"Validator did not mark any pages as failed",
+        ),
+        (
+            {1},
+            set(),
+            set(),
+            r"Validator did not edit any pages",
+        ),
+        (
+            {1},
+            {2},
+            {3},
+            r"Cannot finish validation task. There are not processed pages",
+        ),
+    ),
+)
+def test_validate_user_actions_invalid_states(
+    failed: set,
+    annotated: set,
+    not_processed: set,
+    expected_error_message_pattern: str,
+):
+    with pytest.raises(HTTPException) as excinfo:
+        validate_user_actions(
+            is_validation=True,
+            failed=failed,
+            annotated=annotated,
+            not_processed=not_processed,
+            annotation_user=True,
+            validation_user=True,
+        )
+    assert re.match(expected_error_message_pattern, excinfo.value.detail)
+
+
+def test_create_annotation_task_success():
+
+    with patch("sqlalchemy.orm.Session", spec=True) as mock_session, patch(
+        "annotation.jobs.services.update_user_overall_load"
+    ) as mock_update_user_overall_load, patch(
+        "annotation.models.ManualAnnotationTask"
+    ), patch(
+        "annotation.schemas.tasks.ManualAnnotationTaskInSchema"
+    ):
+
+        db_session = mock_session()
+        create_annotation_task(
+            db_session,
+            ManualAnnotationTaskInSchema(
+                file_id=1,
+                pages={1, 2},
+                job_id=2,
+                user_id=uuid4(),
+                is_validation=True,
+                deadline=None,
+            ),
+        )
+
+        assert db_session.add.call_count == 2
+        mock_update_user_overall_load.assert_not_called()
```

Comment:
For the same reason as the previous comment, this function is called, whereas the mock wasn't called. This should raise a red flag when writing tests.
```suggestion
        mock_update_user_overall_load.assert_called_once()
```

Reply:
Look into `assert_called_once_with()` . You are going to need to save the user_id at the very least.